lib/protocol.js

```diff
@@ -199,10 +204,51 @@ function Protocol (opts) {
 
 util.inherits(Protocol, duplexify)
 
+// extension api
+
+Protocol.prototype.remoteSupports = function (id) {
+  var self = this
+  if (!this._handshook) return
+
+  var offset = id - 64
+  var extension = self._extensions[offset]
+  if (self._available.indexOf(extension) > -1) return true
+  return false
+}
+
+Protocol.prototype.listAvailable = function () {
+  return this._available
+}
+
+Protocol.prototype.send = function (id, buf) {
+  var self = this
+  if (!self.remoteSupports(id)) return
+
+  if (!Buffer.isBuffer(buf)) buf = new Buffer(buf)
+
+  var msg = new Buffer(buf.length + 1)
+  msg[0] = id
```

Comment:
I don't think that we should currently be worried about users having more than 192 extensions, but should we put a thing in the code to check that that doesn't happen? E.g. assert that id < 255?